packages/testing/src/testing.ts

```diff
@@ -86,10 +86,11 @@ export function createTestingPinia({
   createSpy: _createSpy,
 }: TestingOptions = {}): TestingPinia {
   const pinia = createPinia()
+  const state = isVue2 ? Vue2.observable(initialState) : initialState
 
   pinia.use(({ store }) => {
-    if (initialState[store.$id]) {
-      store.$patch(initialState[store.$id])
+    if (state[store.$id]) {
+      store.$patch(state[store.$id])
```

Comment:
@posva 

I noticed the `hmr` `patchObject` method handles reactivity for Vue2. 

```
export function patchObject(
  newState: Record<string, any>,
  oldState: Record<string, any>
): Record<string, any> {
  // no need to go through symbols because they cannot be serialized anyway
  for (const key in oldState) {
    const subPatch = oldState[key]

    // skip the whole sub tree
    if (!(key in newState)) {
      continue
    }

    const targetValue = newState[key]
    if (
      isPlainObject(targetValue) &&
      isPlainObject(subPatch) &&
      !isRef(subPatch) &&
      !isReactive(subPatch)
    ) {
      newState[key] = patchObject(targetValue, subPatch)
    } else {
      // objects are either a bit more complex (e.g. refs) or primitives, so we
      // just set the whole thing
      if (isVue2) {
        set(newState, key, subPatch)
      } else {
        newState[key] = subPatch
      }
    }
  }

  return newState
}
```

Perhaps we should do the same thing for `mergeReactiveObjects` not just in test code. Otherwise it may cause discrepancies in behaviour between testing, HMR and application code. It a nice DX for the `$patch` to consider reactivity principles of Vue 2 and avoid the need for users to consider reactivity constraints for Vue 2. Thoughts?

Reply:
The three cases are really different: HMR (adding, removing new state) and initialState (edge case you highlighted) might need set but usually patch in application code doesn’t (same caveats as Vue, so consistent).

Reply:
@posva fair enough. I pushed changes. `isPlainObject` was not exposed i assume by design so duplicated that method in the testing package also.